apps/src/dance/DanceAiModal.tsx

```diff
@@ -73,14 +75,25 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
   const [currentInputSlot, setCurrentInputSlot] = useState(0);
   const [inputs, setInputs] = useState<string[]>([]);
   const [responseJson, setResponseJson] = useState<string>('');
-  const [responseCall, setResponseCall] = useState<string>('');
   const [responseExplanation, setResponseExplanation] = useState<string>('');
   const [typingDone, setTypingDone] = useState<boolean>(false);
 
   const currentAiModalField = useSelector(
     (state: {dance: DanceState}) => state.dance.currentAiModalField
   );
 
+  useEffect(() => {
+    const currentValue = currentAiModalField?.getValue();
```

Comment:
Talked about this briefly offline - I know this is from a separate PR, but I think generally we want to avoid storing complex class instances in Redux. Typically redux state should just be composed of plain objects/primitive types. 

However, we have a tricky situation here of needing a reference to the original Blockly field in this modal...one thing we could do is just move the reference out to some global reference holder, and instead have the redux field be something like `showAiModal: boolean`? That way Redux can still drive the state that shows/hides the modal, but this component then fetches the modal reference from some other store. This can definitely be a follow-up, but that would let us remove this reference from the redux store.

Reply:
Great point.  Definitely open to improving this, especially as we learn more about which component should own this modal, based on design adjustments.